src/Compilers/CSharp/Portable/Symbols/PublicModel/FunctionPointerTypeSymbol.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class FunctionPointerTypeSymbol : TypeSymbol, IFunctionPointerTypeSymbol
+    {
+        private readonly Symbols.FunctionPointerTypeSymbol _underlying;
+        private IMethodSymbol? _lazySignatureSymbol;
```

Comment:
>private IMethodSymbol? _lazySignatureSymbol; [](start = 8, length = 44)

I am not sure if this cache really needed. I think each symbol already caches its public representation